src/backend/utils/misc/queryenvironment.c

```diff
@@ -67,6 +67,15 @@ struct QueryEnvironment topLevelQueryEnvData;
 struct QueryEnvironment *topLevelQueryEnv = &topLevelQueryEnvData;
 struct QueryEnvironment *currentQueryEnv = NULL;
 
+static void tsql_setup_queryEnv(QueryEnvironment *queryEnv);
```

Comment:
will it make sense to put BBF changes in a separate file? if not possible due to static functions etc we can use wrappers.
Another possibility is to move most of the code to BBF extension

Reply:
the main issue with moving BBF changes to a separate file is that the QueryEnvironment struct is meant to be private to the `queryenvironment.c` file. Moving everything to the extension would be possible if we move the QueryEnvironment struct into `queryenvironment.h`, making it publicly visible, but it has its own risks if community decides to refactor that code in some way.